### PR TITLE
DAT-36 Allow DXF to be fetched form ilr

### PR DESCRIPTION
**Pull request by rcrichton**

### CODE REVIEW

File: src/server.coffee [hunk 1]
```diff
@@ -108,9 +108,35 @@ bothTrigger = (out, callback) ->
...
+    cert: nullIfEmpty config.getConf()['sync-type']['both-trigger-client-cert']
+    key: nullIfEmpty config.getConf()['sync-type']['both-trigger-client-key']
+    ca: nullIfEmpty config.getConf()['sync-type']['both-trigger-ca-cert']
+
+  out.info "Fetching DXF from ILR #{ilrOptions.url} ..."
+  ilrReq = request.post ilrOptions, (err, res, body) ->
+    if err then return callback err
+    if res.statusCode isnt 200 then return callback new Error "Returned non-200 response code: #{body}"
+
+    console.log res.statusCode, body.toString()
```
Comment (hnnesv) [inline src/server.coffee]: debug console.log?

**Comment by hnnesv** [comment]
@rcrichton LGTM - there's just an extra console.log but feel free to go ahead and merge after removing 👍

**Comment by hnnesv** [comment]
@rcrichton @BMartinos FYI you're both merging into different branches (master and dat-32- respectively). Maybe just liase with each other before merging.

File: config/mediator.json [hunk 2]
```diff
@@ -197,6 +197,9 @@
       "empty": false
     },
     "ilr-to-dhis": {
+      "ilr-url": "http://localhost:8984/CSD",
+      "ilr-doc": "test",
+      "dhis2-version": "v2.19"
```
Comment (hnnesv) [inline config/mediator.json]: (oops missed this earlier) You're missing config defs for these. Martin has the ilr- ones defined though.

File: src/server.coffee [hunk 3]
```diff
@@ -108,9 +108,35 @@ bothTrigger = (out, callback) ->
...
+              <preserveUUIDs value='1'/>
+            </csd:requestParams>"""
+    headers:
+      'Content-Type': 'text/xml'
+    cert: nullIfEmpty config.getConf()['sync-type']['both-trigger-client-cert']
+    key: nullIfEmpty config.getConf()['sync-type']['both-trigger-client-key']
+    ca: nullIfEmpty config.getConf()['sync-type']['both-trigger-ca-cert']
+
+  out.info "Fetching DXF from ILR #{ilrOptions.url} ..."
+  ilrReq = request.post ilrOptions, (err, res, body) ->
```
Comment (hnnesv) [inline src/server.coffee]: (another thing i missed!) It would be good to add an orchestration for this

**Comment by hnnesv** [comment]
@rcrichton sorry missed I few extra things - I only thought about these as I was reading through Martin's PR.

**Comment by rcrichton** [comment]
@hnnesv thanks, seems I forgot to do a bunch of stuff, sorry.

I should have been merging with the dat-32 branch, I've corrected that now.

**Comment by rcrichton** [comment]
@hnnesv ready for re-review

**Comment by hnnesv** [comment]
👍 LGTM

**Pull request merged**